src/nft_pp_verifier.cairo

```diff
@@ -0,0 +1,147 @@
+#[starknet::contract]
+mod NftPpVerifier {
+    use core::array::SpanTrait;
+    use core::option::OptionTrait;
+    use starknet::{ContractAddress, get_caller_address, get_contract_address, ClassHash};
+    use traits::{TryInto, Into};
+
+    use openzeppelin::token::erc721::interface::{
+        IERC721CamelOnlyDispatcher, IERC721CamelOnlyDispatcherTrait
+    };
+    use nft_pp_verifier::interface::verifier::{
+        INftPpVerifier, INftPpVerifierDispatcher, INftPpVerifierDispatcherTrait
+    };
+    use nft_pp_verifier::identity::{IIdentityDispatcher, IIdentityDispatcherTrait};
+
+    #[storage]
+    struct Storage {
+        owner_of_starknet_pp: LegacyMap::<(ContractAddress, u256), u128>,
+        whitelisted_contracts: LegacyMap::<ContractAddress, bool>,
+        admin: ContractAddress,
+        identity_contract: ContractAddress,
+    }
+
+    //
+    // Constructor
+    //
+
+    #[constructor]
+    fn constructor(
+        ref self: ContractState, admin_addr: ContractAddress, identity_addr: ContractAddress
+    ) {
+        self.admin.write(admin_addr);
+        self.identity_contract.write(identity_addr);
+    }
+
+    #[external(v0)]
+    impl NftPpVerifierImpl of INftPpVerifier<ContractState> {
+        fn get_pp(self: @ContractState, id: u128) -> (felt252, starknet::ContractAddress, u256) {
+            let nft_contract = self.get_nft_contract(id);
+            let nft_id = self.get_nft_id(id);
+            // todo: handle domain, for now returns 0
```

Comment:
Actually this will not be the domain but the network (is it a native (Starknet) or an Ethereum avatar for example.